src/components/draw/DrawDirective.js

```diff
@@ -27,889 +28,798 @@ goog.require('ga_permalink');
-
-              // If a file has been created we set the correct id to the layer
-              if (data.adminId && data.adminId != layer.adminId) {
-                layer.adminId = data.adminId;
-                layer.url = data.fileUrl;
-                layer.id = 'KML||' + layer.url;
-              }
+          } else {
+            // Move the popup on the closest coordinate of the click event
+            var coord = clickCoord ?
+                feature.getGeometry().getClosestPoint(clickCoord) :
+                feature.getGeometry().getLastCoordinate();
+            var pixel = map.getPixelFromCoordinate(coord);
 
-              if (!scope.adminShortenUrl) {
-                updateShortenUrl(layer.adminId);
-              }
-            });
-          };
-          var saveDebounced = gaDebounce.debounce(save, 2000, false, false);
+            $rootScope.$broadcast('gaDrawStyleActive', scope.feature, layer,
+                pixel, onClosePopup);
+            scope.$applyAsync();
+          }
+        };
 
+
+        ////////////////////////////////////
+        // create/update the file on s3
+        ////////////////////////////////////
+        var save = function(evt) {
+          if (layer.getSource().getFeatures().length == 0) {
+            //if no features to save, delete the file
+            if (layer.adminId) {
+              scope.statusMsgId = '';
+              gaFileStorage.del(layer.adminId).then(function() {
+                layer.adminId = undefined;
+                layer.url = undefined;
+              });
+            }
+            map.removeLayer(layer);
+            return;
+          }
+          scope.statusMsgId = 'draw_file_saving';
+          var kmlString = gaExportKml.create(layer,
+              map.getView().getProjection());
+          var id = layer.adminId ||
+              gaFileStorage.getFileIdFromFileUrl(layer.url);
+          gaFileStorage.save(id, kmlString,
+              'application/vnd.google-earth.kml+xml').then(function(data) {
+            scope.statusMsgId = 'draw_file_saved';
+
+            // If a file has been created we set the correct id to the layer
+            if (data.adminId && data.adminId != layer.adminId) {
+              layer.adminId = data.adminId;
+              layer.url = data.fileUrl;
+              layer.id = 'KML||' + layer.url;
+            }
+          });
+        };
+        var saveDebounced = gaDebounce.debounce(save, 2000, false, false);
+
+
+        var dereg = [
           $rootScope.$on('$translateChangeEnd', function() {
             if (layer) {
               layer.label = $translate.instant('draw_layer_label');
             }
-          });
+          })
+        ];
 
-          // Remove interactions on destroy
-          scope.$on('$destroy', function() {
-            deactivate();
+        // Remove interactions on destroy
+        scope.$on('$destroy', function() {
+          deactivate();
+          dereg.forEach(function(item) {
```

Comment:
dereg only ever contaions one element. So we could use it directly, instead of putting it in an array. Minor.